contrib/better-defaults/keybindings.el

```diff
@@ -14,3 +14,5 @@
 
 ;; emacs state bindings
 (define-key evil-emacs-state-map (kbd "C-o") 'evil-execute-in-normal-state)
+(add-hook 'dired-mode-hook (lambda ()
+                             (local-set-key (kbd "M-o") 'dired-display-file)))
```

Comment:
My guess is that `dired-mode-map` is only created when you actually run `dired` command.